File: app/src/main/java/com/example/onlinemarket/Order.java
package com.example.onlinemarket;

import com.example.onlinemarket.Product;

public class Order {

    public String _id;
    public Product product;
    public Integer quantity;
    public String userId;

    public Order(
            String _id,
            Product product,
            Integer quantity,
            String userId
    ){
        this._id = _id;
        this.product = product;
        this.quantity = quantity;
        this.userId = userId;
    }
}

File: server/api/models/order.js
// import mongoose for mongodb support
// This is used to create a mongo schema
const mongoose = require('mongoose');

// Create the schema
const orderSchema = mongoose.Schema({
    // Define the schema attributes
    _id: mongoose.Schema.Types.ObjectId,
    product: { type: mongoose.Schema.Types.ObjectId, ref: 'Product', required: true },
    quantity: { type: Number, default: 1 },
    userId: { type: String, required: true }
});

// export the module for use in other files
module.exports = mongoose.model('Order', orderSchema);